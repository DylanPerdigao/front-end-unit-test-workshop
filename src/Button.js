import React, { PureComponent } from 'react';
import PropTypes from "prop-types";

export default class Button extends PureComponent {
    propTypes = {
        text: PropTypes.string.isRequired,
        onClick: PropTypes.function
    };

    render() {
        const {text, capitalize, onClick} = this.props;
        let label = text;

        if (capitalize) {
            label = text.slice(0, 2).toUpperCase() + text.slice(2);
        }

        return (
            <button onClick={onClick} type="button">{label}</button>
        );
    }
}
