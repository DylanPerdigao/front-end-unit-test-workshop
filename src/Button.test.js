import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";
import toJSON from "enzyme-to-json";

describe("render", () => {
    it("renders the given text", () => {
        const wrapper = shallow(<Button text="Jump" />);

        expect(wrapper.text()).toEqual("Jump");
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
