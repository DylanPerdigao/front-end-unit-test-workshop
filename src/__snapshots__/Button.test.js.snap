// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders the given text 1`] = `
<button
  type="button"
>
  Jump
</button>
`;
